modules/embedding-microshift-into-rpm-ostree-image.adoc

```diff
@@ -0,0 +1,212 @@
+      "baseurl": "https://cdn.redhat.com/content/dist/layered/rhel8/x86_64/fast-datapath/os",
+      "gpgkey": "-----BEGIN PGP PUBLIC KEY BLOCK----- [...] -----END PGP PUBLIC KEY BLOCK-----\n",
+      "rhsm": true,
+      "check_gpg": true
+    }
+  ]
+}
+----
+
+Restart the `osbuild-composer.service` to load the override repository definition:
+
+[source,terminal]
+----
+$ sudo systemctl restart osbuild-composer.service
+----
+
+== Adding MicroShift to an Image Builder blueprint
+=== Adding the MicroShift service
+To build an {op-system-ostree} image with MicroShift, simply add the MicroShift
+RPM package to a blueprint and enable the MicroShift service. A minimal Image
+Builder blueprint would look as follows:
+
+[source,terminal]
+----
+name = "minimal-microshift"
+
+description = ""
+version = "0.0.1"
+modules = []
+groups = []
+
+[[packages]]
+name = "microshift"
+version = "{ocp-version}.0-1"
+
+[[packages]]
+name = "redhat-release"
+version = "*"
+
+[customizations.services]
+enabled = ["microshift"]
+----
+
+If your use case requires running the `oc` CLI tool locally on your MicroShift
+machine, further add the `openshift-clients` package.
+If your use case requires using MicroShift's "https://[auto-deployed manifests]"
+feature with manifests stored on a remote git repository, further add the `git`
+package.
+
+You can further customize your image, e.g. to add users, firewall rules, or
+kernel arguments. Refer to the documentation on
+https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/composing_installing_and_managing_rhel_for_edge_images/composing-a-rhel-for-edge-image-using-image-builder-command-line_composing-installing-managing-rhel-for-edge-images#image-customizations_composing-a-rhel-for-edge-image-using-image-builder-command-line[supported image customizations].
+
+=== Adding the MicroShift container images
+Image Builder supports building rpm-ostree images with embedded container
+images. Embedded container images are immediately available to container
+engines like podman and cri-o after booting, without having to pull these
+images over the network from a container registry.
+
+A container image is embedded by adding a reference to it to the Image
+Builder blueprint like this:
+
+[source,terminal]
+----
+[[containers]]
+source = "<image_pullspec_with_tag_or_digest>"
+----
+
+To embed MicroShift's container images, you need to know the exact list of
+container image references used by the MicroShift version you are embedding.
+You can obtain this list from the `microshift-release-info` RPM package
+matching the version of the `microshift` RPM in your blueprint using the
+following procedure.
+
+.Procedure
+
+. Download the `microshift-release-info` RPM package matching your MicroShift version using the following command:
++
+[source,terminal]
+----
+$ dnf download microshift-release-info-{ocp-version}.0-1.el{rhel-major}
+----
+
+. Exctract the release info by entering the following command:
```

Comment:
. Extract the release info by running the following command:

Reply:
Updated.